### PR TITLE
Adds Non Null Input fields cannot be inaccessible

### PR DESCRIPTION
**Pull request by PascalSenn**

### CODE REVIEW

**Comment by PascalSenn** [comment]
Do we automatically remove a field that references an inaccessible type, or must the field be explicitly declared as inaccessible to avoid a composition error?

File: spec/Section 4 -- Composition.md [hunk 1]
```diff
@@ -18,4 +18,119 @@ run in sequence to produce the composite execution schema.
...
+
+NON_NULL_INPUT_FIELD_IS_INACCESSIBLE
+
+**Formal Specification**
+
+- Let {fields} be the set of all fields of the input types
+- For each {field} in {fields}:
+  - If {field} is a non-null input field:
+    - {field} must not be declared as `@inaccessible`
+
```
Comment (PascalSenn) [inline spec/Section 4 -- Composition.md:33]: @michaelstaib  should we have a generalised rule for require?

File: spec/Section 4 -- Composition.md [hunk 2]
```diff
@@ -18,4 +18,119 @@ run in sequence to produce the composite execution schema.
...
+  field1: String! @inaccessible
+  field2: Input2
+}
+
+input Input2 {
+  field3: String
+}
+```
+
+#### Input Fields cannot reference be inaccessible type
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:78]:
```suggestion
#### Input Fields cannot reference inaccessible type
```

File: spec/Section 4 -- Composition.md [hunk 3]
```diff
@@ -18,4 +18,119 @@ run in sequence to produce the composite execution schema.
...
+**Formal Specification**
+
+- Let {fields} be the set of all fields of the input types
+- For each {field} in {fields}:
+  - Let {namedType} be the named type that {field} references
+  - {namedType} must not be declared as `@inaccessible`
+
+**Explanatory Text**
+
+In a composed schema, a field within a input type must only reference types that are exposed.
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:93]:
```suggestion
In a composed schema, a field within an input type must only reference types that are exposed. 
```

File: spec/Section 4 -- Composition.md [hunk 4]
```diff
@@ -18,4 +18,119 @@ run in sequence to produce the composite execution schema.
...
+  field1: String!
+  field2: Input2!
+}
+
+input Input2 @inaccessible {
+  field3: String
+}
+```
+
+
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:135]:
```suggestion
```

**Review by glen-84 — CHANGES_REQUESTED** [review]

**Comment by PascalSenn** [comment]
Closed by #93

**Pull request closed without merging**